### ISSUE TITLE
Icon for Visual Studio Code (vscode)

### ISSUE THREAD

**Issue by edgard** [numixproject/numix-icon-theme-circle#2964]
vS code has been release today as 1.0 and the icon is not being used for the deb package. I noticed that there is already a visualstudiocode.svg, but the .desktop file is as follows:

/usr/share/applications/code.desktop
```
[Desktop Entry]
Name=Visual Studio Code
Comment=Code Editing. Redefined.
GenericName=Text Editor
Exec=/usr/bin/code %U
Icon=code
Type=Application
StartupNotify=true
Categories=Utility;TextEditor;Development;IDE;
MimeType=text/plain;
```

Suppose it needs a new symlink?

**Comment by palob** [comment]
Hmm, just ``code`` is quite the generic icon name.

**Comment by edgard** [comment]
heh, yeah. but maybe because it's so generic, it seems like nobody dared to use before. :D

**Comment by Foggalong** [comment]
@edgard That's not a good thing, that's a bad thing. Generic icon names are likely to cause conflicts. I'd wager just symlink this to the generic text editor icon. It's better than nothing and I'm not sure we can comfortably fix this as just a VSCode icon.

**Comment by edgard** [comment]
While I also do not like its called 'code' (even the cli), it's how it is right now. They oficially packaged like that. :)

**Comment by wa4557** [comment]
Is it possible to include a "fix" for that in the hardcode fixer script?

**Comment by Foggalong** [comment]
@edgard Yeah, they might have packaged it like that but it doesn't mean they've done it right. Lots of devs use `icon.svg` as an icon name but it's a pain for the same reason. We can't give it a specific icon because they used too generic a name which others are likely to also use. Please report it to them and ask for a fix.

@wa4557 wecould, yeah

**Comment by icasdri** [comment]
I'm on a different distribution (namely Arch), that is currently packaging (unofficially by community members) with a different `Icon` line. So maybe hardcode fix and add icon for something like this while still generically linking "code" to text editor icon?
<pre>
[Desktop Entry]
Categories=Development;
Exec=/usr/bin/visual-studio-code %f
<b>Icon=visual-studio-code</b>
Terminal=false
Type=Application
StartupWMClass=code
Name=Visual Studio Code
Comment=Editor for building and debugging modern web and cloud applications
Categories=Development;WebDevelopment;IDE;
StartupNotify=true
</pre>

**Comment by ChuckDaniels87** [comment]
This should be reported [upstream](https://github.com/Microsoft/vscode/tree/master/scripts).  I agree that `code` is not a proper icon name. Each distro is packaging it with a different name in its `desktop file`. 

The [original code.desktop](https://github.com/Microsoft/vscode/blob/master/resources/linux/code.desktop) has a placeholder called `@@NAME@@`. I really can't see the reason of this design.

**Comment by palob** [comment]
People keep reporting this. Maybe it's time to change our stance? We'll have to make sure there is no action/status/... icon named ``code(-...)``.

To people facing this issue: you can fix it by copying the ``.desktop`` file over to ``~/.local/share/applications`` and editing the icon line inside to ``Icon=vscode``.

**Comment by sajidali** [comment]
@palob Your fix suggestion worked perfectly fine.

**Comment by ghost** [comment]
Changing the .desktop file to `Icon=vscode` Does not work for Numix Square icons, but does work for Numix Circle icons.

**Comment by palob** [comment]
We don't have a vscode icon for Square yet.

**Comment by tluanga34** [comment]
No fix till today :(

**Comment by bilelmoussaoui** [comment]
Hey, it's already fixed. The icon name must be replaced manually to vscode or visualstudiocode. As the `code` icon name is used by other applications like Gedit...

**Comment by tluanga34** [comment]
Gedit has an icon name gedit

[Desktop Entry]
Name=Text Editor
Comment=Edit text files
Exec=gedit %U
Terminal=false
Type=Application
StartupNotify=true
Icon=gedit
Categories=GNOME;GTK;Utility;TextEditor;
X-GNOME-DocPath=gedit/gedit.xml
X-GNOME-FullName=Text Editor
X-GNOME-Bugzilla-Bugzilla=GNOME
X-GNOME-Bugzilla-Product=gedit
X-GNOME-Bugzilla-Component=general
X-GNOME-Bugzilla-Version=3.20.2
X-GNOME-Bugzilla-ExtraInfoScript=/usr/share/gedit/gedit-bugreport.sh
Actions=new-window;new-document;
Keywords=Text;Editor;Plaintext;Write;
Keywords[oc]=Tèxte;Editor;Tèxte brut;Escriure;
X-Ubuntu-Gettext-Domain=gedit

X-AppStream-Ignore=true

[Desktop Action new-window]
Name=Open a New Window
Exec=gedit --new-window

[Desktop Action new-document]
Name=Open a New Document
Exec=gedit --new-document

**Comment by Foggalong** [comment]
@tluanga34 As with all these things it depends on the packaging source, though I don't know off hand who's responsible for the `Icon=code` gaffe there. This remains a wontfix for the [same reasons](https://github.com/numixproject/numix-core/issues/2964#issuecomment-210524377) that should be instead fixed by VS upstream. There's no shortage of appropriate names they could use (`vsc`, `vscode`, `visual-studio-code`, etc) that aren't genericised. I'm adding a wontfix label to emphasise this won't be taken further on our end.

**Comment by colemickens** [comment]
@Foggalong Will you include aliases for some of the names that *are* used in other packaging?

I'm still trying to track down how `vscode` the repo is packaged, as the desktop entry in the source repo just has a placeholder for the Icon: `Icon=@@ICON@@`.

But, for example, the Arch AUR package for `visual-studio-code-insiders` sets the icon to `visual-studio-code-insiders`, which, I thought for a while, had been fulfilled by numix. Or maybe that memory is from before I switched to the `-insiders` package...

**Comment by Foggalong** [comment]
@colemickens Yeah of course! We tried to get some more information about VSCode Insiders in #3902 to see whether it was a new icon or just a symlink, but we were waiting on a response from another user. If you have that info please post it there.

**Comment by beruic** [comment]
@colemickens, the placeholder seems to be replaced [here](https://github.com/Microsoft/vscode/blob/master/build/gulpfile.vscode.linux.js#L41). I'm looking into making a pull request to fix this issue in the build of VS Code it self.

**Comment by beruic** [comment]
My PR was rejected. See the closing comment [here](https://github.com/Microsoft/vscode/pull/62650#issuecomment-443761684).

**Comment by Foggalong** [comment]
@beruic Replied there, hopefully to shed some light on why the name can't just be covered.

**Comment by Foggalong** [comment]
As an update, they've agreed to look into changing the icon name of VS Code to `com.visualstudio.code` to match Free Desktop standards which will fix this issue for us and other themes. Not confirmed yet, but looks positive :)

**Comment by Tyriar** [comment]
Thanks for your patience everyone, I just merged this in so from VS Code v1.33.0 the icon will be `com.visualstudio.code` for stable, `com.visualstudio.code.insiders` for insiders and `com.visualstudio.code.oss` for the OSS build.

**Comment by bilelmoussaoui** [comment]
@Tyriar Thanks for looking into this. Can we get the desktop file renamed too? As the future of packages in Linux is Snap's and Flatpak's and they enforcing using a RDNN name for all the exported files. Having that done would be pretty nice, I will open a new issue about that later with all the links needed for docs.

**Comment by Foggalong** [comment]
@Tyriar Thanks so much for sorting this! I'll make sure we've got all those in the master branch so the icons are ready in place when the update lands.

@bilelmoussaoui Peddle your Flatpak wares elsewhere :wink:
